### PR TITLE
Don't allow implicit conversions on prefixes of type selections

### LINKED ISSUE #19888: Implicit conversions can add extension type members
## Compiler version

3.4.0

## Minimized code

```Scala
object Main{
  object Foo
  object Bar{ type Qux = Unit }


  implicit def conv(f: Foo.type): Bar.type = Bar

  def B: Bar.type = Bar

  def main(args: Array[String]): Unit = {
    val x: Foo.Qux = null
    val y: B.Qux = null
    println(x)
    println(y)
  }  
}
```

## Output

```scala
null
null
```

## Expectation

I would expect `A.Qux` to be a compilation error, because `A` does not have a member `type Qux`, without the implicit conversion. 

I would expect `B.Qux` to be a compilation error, because `B` is not a stable path prefix for the `B.Qux` path dependent type

The equivalent code is a compilation error in Scala 2.x. It definitely seems weird that implicit conversions and other non-stable methods can be used in the path dependent type's path prefix!

TBH not sure if this is bug or feature, but @odersky mentioned it might be a bug, so filing it here for posterity.

### PR DESCRIPTION
**Pull request by odersky**
Fixes #19888

### CODE REVIEW

File: tests/neg/i11994.check [hunk 1]
```diff
@@ -1,26 +1,12 @@
--- [E008] Not Found Error: tests/neg/i11994.scala:2:28 -----------------------------------------------------------------
-2 |implicit def foo[T <: Tuple.meow]: Unit = ??? // error
```
Comment (noti0na1) [inline tests/neg/i11994.check:2]: I think `implicit def foo[T <: Tuple.meow]: Unit` should still produce an error?

File: tests/neg/i11994.check [hunk 2]
```diff
@@ -1,26 +1,12 @@
...
-  |                          
-  |                           Run with both -explain-cyclic and -Ydebug-cyclic to see full stack trace.
--- [E008] Not Found Error: tests/neg/i11994.scala:3:18 -----------------------------------------------------------------
+-- [E046] Cyclic Error: tests/neg/i11994.scala:3:18 --------------------------------------------------------------------
 3 |given [T <: Tuple.meow]: Unit = ??? // error
-  |            ^^^^^^^^^^
-  |            type meow is not a member of object Tuple.
-  |            Extension methods were tried, but the search failed with:
+  |                  ^
+  |                  Cyclic reference involving method given_Unit
```
Comment (noti0na1) [inline tests/neg/i11994.check:4]: `Tuple.meow` is always illegal. This message is more useless than before and not showing this fact.

**Review by noti0na1 — CHANGES_REQUESTED** [review]
The fix in this PR may hide some illegal type selections.

File: tests/neg/i11994.check (hunk 2 above)
Comment (odersky) [reply tests/neg/i11994.check]: Indeed. It was a problem in the error reporting that caused the real error to be hidden.

**Review by noti0na1 — APPROVED** [review]
LGTM

**Pull request merged**